pkg/reckoner/plot.go

```diff
@@ -138,6 +138,19 @@ func (c Client) TemplateAll(templateOutputDir string) (fullOutput string, err er
 
 		fullOutput = fullOutput + out
 	}
+
+	// if the user has specified a RepoURL, they probably also
+	// specified other things. Most required things receive a default
+	// and a warning when not found in the config file. This
+	// maybe could use some improvement
+	if len(c.CourseFile.GitOps.ArgoCD.Spec.Source.RepoURL) > 0 {
+		err = c.WriteArgoApplications(templateOutputDir)
+		if err != nil {
+			color.Red(err.Error())
+			os.Exit(1)
+		}
+	}
+
```

Comment:
I'm noticing that this only runs when we `template all` - but if I'm not mistaken, we can still do --output-dir when we only template a single chart. 

I think this should be triggered closer to the actual template function, perhaps in the TemplateRelease function, rather than here. What are your thoughts?

Reply:
Good catch! I can't believe I didn't notice this during development/testing. Thanks very much for pointing that out. I've moved it to the appropriate function (`TemplateRelease()`) where it should've been all along :)